runtime/src/bank/check_transactions.rs

```diff
@@ -184,6 +188,54 @@ impl Bank {
     }
 }
 
+/// If the message uses a durable nonce, return the pubkey of the nonce account
+fn get_durable_nonce(message: &impl SVMMessage) -> Option<&Pubkey> {
+    // Must be first instruction for below code to work.
+    const_assert_eq!(NONCED_TX_MARKER_IX_INDEX, 0);
```

Comment:
Not too sure how I feel about it, my first thought is this assertion might belong to where the const is defined, to reenforce the nonced instruction requirement. Call site uses that const just need to trust it, and use `get(NONCED_TX_MARKER_IX_INDEX)` instead of `next()`. wdyt?

Reply:
Yeah I can just do `nth` like I did for the other one! Not sure why I did it this way here but using `nth` below 😕